cli/compose/loader/loader_test.go

```diff
@@ -799,3 +930,89 @@ type servicesByName []types.ServiceConfig
 func (sbn servicesByName) Len() int           { return len(sbn) }
 func (sbn servicesByName) Swap(i, j int)      { sbn[i], sbn[j] = sbn[j], sbn[i] }
 func (sbn servicesByName) Less(i, j int) bool { return sbn[i].Name < sbn[j].Name }
+
+func TestLoadExpandedPortFormat(t *testing.T) {
+	config, err := loadYAML(`
+version: "3.1"
+services:
+  web:
+    image: busybox
+    ports:
+      - "80-82:8080-8082"
+      - "90-92:8090-8092/udp"
+      - "85:8500"
+      - 8600
+      - protocol: udp
+        target: 53
+        published: 10053
+      - mode: host
+        target: 22
+        published: 10022
+`)
+	assert.NoError(t, err)
+
+	expected := []types.ServicePortConfig{
+		{
+			Mode:      "ingress",
+			Target:    8080,
+			Published: 80,
+			Protocol:  "tcp",
+		},
+		{
+			Mode:      "ingress",
+			Target:    8081,
+			Published: 81,
+			Protocol:  "tcp",
+		},
+		{
+			Mode:      "ingress",
+			Target:    8082,
+			Published: 82,
+			Protocol:  "tcp",
+		},
+		{
+			Mode:      "ingress",
+			Target:    8090,
+			Published: 90,
+			Protocol:  "udp",
+		},
+		{
+			Mode:      "ingress",
+			Target:    8091,
+			Published: 91,
+			Protocol:  "udp",
+		},
+		{
+			Mode:      "ingress",
+			Target:    8092,
+			Published: 92,
+			Protocol:  "udp",
+		},
+		{
+			Mode:      "ingress",
+			Target:    8500,
+			Published: 85,
+			Protocol:  "tcp",
+		},
+		{
+			Mode:      "ingress",
+			Target:    8600,
+			Published: 0,
+			Protocol:  "tcp",
+		},
+		{
+			Target:    53,
+			Published: 10053,
+			Protocol:  "udp",
+		},
+		{
+			Mode:      "host",
+			Target:    22,
+			Published: 10022,
+		},
+	}
+
+	assert.Equal(t, 1, len(config.Services))
+	assert.Equal(t, len(expected), len(config.Services[0].Ports))
```

Comment:
I would remove this assertion. It's covered by the next line, and it'll be easier to see the failure if the error shows the full diff, instead of just `x != y`